pyodide-build/pyodide_build/pywasmcross.py

```diff
@@ -672,12 +678,9 @@ def handle_command(
     is_link_cmd = get_library_output(line) is not None
 
     if line[0] == "gfortran":
-        if "-dumpversion" in line:
-            sys.exit(subprocess.run(line).returncode)
         tmp = replay_f2c(line)
-        if tmp is None:
-            sys.exit(0)
-        line = tmp
+        if tmp is not None:
+            line = tmp
```

Comment:
Can you explain this change? In particular we used to quit out here if `replay_f2c` returns `None` but now we don't? Why? When does this happen? What is the subprocess call below going to do in this case?

Also, we should switch `handle_command` to return an `int` since in `compile_main` it says `sys.exit(handle_command(args, build_args))` we can just `return 0` or `return returncode` in here.

Reply:
As mentioned in another comment, meson runs gfortran multiple times during the compiler check phase, so sometimes replay_f2c doesn't do anything. So I changed it to run gfortran normally in that case.

Reply:
> Also, we should switch handle_command to return an int since in compile_main it says sys.exit(handle_command(args, build_args)) we can just return 0 or return returncode in here.

Right, thanks for the pointer.

Reply:
Would the following work:
```
tmp = replay_f2c(line)
if tmp is None:
   # No source file, it's a query for information about the compiler. Pretend we're
   # gfortran by letting gfortran handle it
   return subprocess.run(line).returncode
line = tmp
```

Or is it important that we are calling `handle_command_generate_args` on it? I think it would help to have some comments here since this is a really confusing execution path.

Reply:
I think your code will work. Thanks for the suggestion. 

> since this is a really confusing execution path.

I agree. Actually `pywasmcross` is becoming more complicated as we add more symlinks... I'll try do to some refactorings when I have bandwidth.